addons/stock/models/stock_inventory.py

```diff
@@ -204,15 +232,42 @@ def _get_inventory_lines_values(self):
                 product_data[void_field] = False
             product_data['theoretical_qty'] = product_data['product_qty']
             if product_data['product_id']:
-                product_data['product_uom_id'] = Product.browse(product_data['product_id']).uom_id.id
+                product = Product.browse(product_data['product_id'])
```

Comment:
So here you are re-browsing a record you have fetched in a sql query 10 lines above... can't you adapt the query to get the uom_id at the same time ?